metals/src/main/scala/scala/meta/internal/worksheets/WorksheetProvider.scala

```diff
@@ -220,9 +224,16 @@ class WorksheetProvider(
       input = path.toInputFromBuffers(buffers)
     } yield mdoc.evaluateWorksheet(input.path, input.value)
   }.map { worksheet =>
+    val toPublish = worksheet
+      .diagnostics()
+      .iterator()
+      .asScala
+      .filterNot(_.position().isNone)
```

Comment:
Should we not at least try to publish those diagnostics at the start? Will it not just hide it from the user? We could show it at the start of the file maybe?

Reply:
Good point, I changed these diagnostics to display in the beginning of the file instead.